pkg/messaging/v1/util.go

```diff
@@ -516,6 +518,13 @@ func ProtobufToJSON(message protoreflect.ProtoMessage) ([]byte, error) {
 // WithCustomGRPCMetadata applies a metadata map to the outgoing context metadata.
 func WithCustomGRPCMetadata(ctx context.Context, md map[string]string) context.Context {
 	for k, v := range md {
+		if strings.EqualFold(k, ContentTypeHeader) ||
```

Comment:
Nit, but here should have used `==` and not `EqualFold`, and mute the linter. `EqualFold` can be a lot slower as it's unicode-aware, and these strings are never Unicode...

Reply:
== will not work because it does not ignorecase. Unless we use Lower() the string. So, is it better to use EqualFold() or == with Lower()